src/components/auth/service/auth.service.js

```
@@ -0,0 +1,172 @@
+'use strict';
+
+const { uuid } = require('uuidv4');
+const bcrypt = require('bcrypt');
+
+const UsersService = require('../../users/users.service');
+const EmailService = require('./email.service');
+const JwtService = require('./jwt.service');
+const TokenService = require('./token.service');
+
+const ApplicationErrors = require('../../exceptions/application.errors');
+
+const validateEmail = (email) => {
+  if (!email) {
+    throw ApplicationErrors.BadRequest('Email should be added');
+  }
+
+  const emailPattern = /^[\w.-]+@[\w.-]+\.\w+$/;
+
+  if (!emailPattern.test(email)) {
+    throw ApplicationErrors.BadRequest('Email should be valid');
+  }
+};
+
+const validatePassword = (password) => {
+  if (!password) {
+    throw ApplicationErrors.BadRequest('Password is required');
+  }
+
+  if (password.length < 8) {
+    throw ApplicationErrors.BadRequest(
+      'The password length should not be less than 8 characters',
+    );
+  }
+};
+
+const register = async (email, password) => {
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
```

The `validateEmail` and `validatePassword` functions throw exceptions instead of returning errors. Modify these functions to return error messages that can be collected in the `errors` object without throwing exceptions immediately.